File: go/data_structure/1_chapter/extract_root.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var n, k int
	var err error
	for err == nil {
		fmt.Println("please input two number: ")
		_, err = fmt.Scan(&n, &k)
		fmt.Println("the result is :" + strconv.Itoa(G(n, k)))
	}
}

//if k == 0; the function is extract rooting function
func G(n, k int) int {
	if n < 1 {
		return k
	} else {
		return G(n-2*k-1, k+1)
	}
}

File: cpp/data_structures/one_chapter/extract_root.c
#include <iostream>

using namespace std;

int G(int n, int k) {           //if k == 0; the function is extract rooting function
    return n < 1 ? k : G(n - 2 * k - 1, k + 1);
}

int main(int argc, char** argv)
{
    int n, k;
    while(cout << "please input one number: ", cin >> n >> k)
        cout << "the result is :" << G(n, k) << endl;
    
    return 0;
}
